apps/prairielearn/src/ee/lib/validateHTML.ts

```diff
@@ -281,6 +281,12 @@ function checkIntegerInput(ast: DocumentFragment | ChildNode): ValidationResult
           break;
         // string inputs are valid as strings, and these don't affect other tags, so no validation required
         case 'correct-answer':
+          if (val.match(mustacheTemplateExtractorRegex)) {
+            errors.push(
+              "pl-integer-input: correct-answer must be a static integer, not dynamically generated in `server.py`. If the correct answer depends on internal parameters, simply set `data['correct_answers']` accordingly and remove this attribute.",
```

Comment:
Nit, workshopping this a bit:

```suggestion
              "pl-integer-input: correct-answer attribute value must not be a Mustache template. If the correct answer depends on dynamic parameters, set `data['correct_answers']` accordingly in `server.py` and remove this attribute.",
```